lib/commands/util/upgrade-project/hosted-skill-helper.js

```diff
@@ -29,10 +29,13 @@ function checkIfDevBranchClean(gitClient) {
     if (statusOutput.toString()) {
         throw new CLiError(`Commit the following files in the dev branch before upgrading project:\n${statusOutput}`);
     }
-    const countOutput = gitClient.countCommitDifference('origin/dev', 'dev');
-    const counter = countOutput.toString().replace(/\D/g, '');
-    if (stringUtils.isNonBlankString(counter) && counter !== '0') {
-        throw new CLiError('Upgrade project failed, as your dev branch is ahead of the remote.\nPlease follow the project upgrade instruction from '
+    let diffCountWithOrigin = gitClient.countCommitDifference('origin/dev', 'dev');
+    diffCountWithOrigin = diffCountWithOrigin.toString().replace(/\D/g, '');
+    let diffCountWithMaster = gitClient.countCommitDifference('master', 'dev');
+    diffCountWithMaster = diffCountWithMaster.toString().replace(/\D/g, '');
+    if ((stringUtils.isNonBlankString(diffCountWithOrigin) && diffCountWithOrigin !== '0')
+    || (stringUtils.isNonBlankString(diffCountWithMaster) && diffCountWithMaster !== '0')) {
+        throw new CLiError('Upgrade project failed. Please follow the project upgrade instruction from '
         + 'https://github.com/alexa-labs/ask-cli/blob/develop/docs/Upgrade-Project-From-V1.md#upgrade-steps to change to ask-cli v2 structure.');
```

Comment:
Is it possible to do the error message for two cases separately? Similar to before, it tells user that the dev branch is ahead. Now we will also have the case that dev is ahead of master. Both cases can be pointed to the instruction.

Reply:
You mean except instruction, also let user know how many different commits in this two conditions?

Reply:
Since user might not realize that it's v1 ask deploy that do the work to merge dev to master, if we only tell them to follow the instruction, users might not connect this logical gap to the current failure.
The most transparent way is to let users know the direct (closest) reason, either it's because they missed the local dev branch change, or it's because dev and master not in sync. ANd then they will understand they need to follow the instruction to re-deploy v1 's changes

Reply:
How about an error message like this “[Error]: Upgrade project failed. Your branch is ahead of mater by 1 commit(s), Please follow the project upgrade instruction from https://github.com/alexa-labs/ask-cli/blob/develop/docs/Upgrade-Project-From-V1.md#upgrade-steps to clean your working branch before upgrading project."